Missing a Windows guard

```
20:03:35 ======================================================================
20:03:35 ERROR: test_trace_of_script (test_jit.TestScript)
20:03:35 ----------------------------------------------------------------------
20:03:35 Traceback (most recent call last):
20:03:35   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test\test\test_jit.py", line 2633, in test_trace_of_script
20:03:35     self.assertEqual(3, use(torch.ones(1, dtype=torch.float)))
20:03:35 RuntimeError: NYI: fuser is not supported on Windows.
20:03:35 
20:03:35 ----------------------------------------------------------
```

--- a/test/test_jit.py
+++ b/test/test_jit.py
@@ -2613,22 +2613,26 @@ class TestScript(TestCase):
         with self.assertRaisesRegex(RuntimeError, 'called recursively involving'):
             M()
 
+    @unittest.skipIf(IS_WINDOWS, "NYI: fuser support for Windows")
     def test_trace_of_script(self):
         @torch.jit.script
         def foo(a, c):
-            b = 0
-            if a == 0:
-                b = 1
+            b = 0.0
+            if a == 0.0:
+                b = 1.0
             return b + c
 
-        a = torch.ones(1, dtype=torch.long)
+        a = torch.ones(1, dtype=torch.float)
 
-        @torch.jit.trace(torch.zeros(1, dtype=torch.long))
+        @torch.jit.trace(torch.zeros(1, dtype=torch.float))
         def use(b):
-            return foo(b - 1, a) + 1
+            return foo(b - 1.0, a) + 1.0
 
-        self.assertEqual(3, use(torch.ones(1, dtype=torch.long)))
-        self.assertEqual(2, use(torch.zeros(1, dtype=torch.long)))
+        # test we propagated shapes through the function
+        self.assertTrue("Dynamic" not in str(use.graph))
+
+        self.assertEqual(3, use(torch.ones(1, dtype=torch.float)))
+        self.assertEqual(2, use(torch.zeros(1, dtype=torch.float)))
 
     def test_if_define(self):
         @torch.jit.script

--- a/torch/csrc/jit/graph_executor.cpp
+++ b/torch/csrc/jit/graph_executor.cpp
@@ -216,9 +216,29 @@ private:
     auto input_values = fmap(input_vars, [&](const Variable& v) {
       return tracer::getValueTrace(state, v);
     });
+
+    ArgumentSpec spec(autograd::GradMode::is_enabled(), inputs);
     input_vars.clear(); // don't hold inputs during execution
     auto outputs = runFallback(std::move(inputs));
-    auto output_values = script::inlineCallTo(*state->graph, *this->graph, input_values);
+
+    auto all_dynamic = [](const at::ArrayRef<Value*> xs) {
+      for(Value* x : xs) {
+        if(x->type()->kind() != TypeKind::DynamicType)
+          return false;
+      }
+      return true;
+    };
+    // Traces always have types propagated through them, so we make sure to
+    // also propagate types through the graph we are inserting here.
+    // However, this->graph itself may already have been generated with
+    // tracing and so we only do the type propgation if no concrete types have
+    // been set.
+    auto local_graph = this->graph;
+    if(all_dynamic(local_graph->inputs()) && all_dynamic(local_graph->outputs())) {
+      local_graph = this->graph->copy();
+      PropagateInputShapes(*local_graph, spec);
+    }
+    auto output_values = script::inlineCallTo(*state->graph, *local_graph, input_values);
 
     for(size_t i = 0; i < outputs.size(); ++i) {
       tracer::setValueTrace(state, outputs[i], output_values[i]);

--- a/torch/onnx/utils.py
+++ b/torch/onnx/utils.py
@@ -182,7 +182,6 @@ def _export(model, args, f, export_params=True, verbose=False, training=False,
     graph, params, torch_out = _model_to_graph(model, args, f, verbose,
                                                training, input_names,
                                                output_names, aten)
-
     # TODO: Don't allocate a in-memory string for the protobuf
     from torch.onnx.symbolic import _onnx_opset_version
     defer_weight_export = export_type is not ExportTypes.PROTOBUF_FILE

